hudi-utilities/src/test/java/org/apache/hudi/utilities/functional/TestHoodieMultiTableDeltaStreamer.java

```diff
@@ -177,6 +182,183 @@ public void testMultiTableExecutionWithKafkaSource() throws IOException {
+    List<TableExecutionContext> executionContexts = streamer.getTableExecutionContexts();
+    TypedProperties properties = executionContexts.get(1).getProperties();
+    properties.setProperty("hoodie.deltastreamer.schemaprovider.source.schema.file", dfsBasePath + "/source_uber.avsc");
+    properties.setProperty("hoodie.deltastreamer.schemaprovider.target.schema.file", dfsBasePath + "/target_uber.avsc");
+    properties.setProperty("hoodie.deltastreamer.source.kafka.topic", topicName2);
+    properties.setProperty("hoodie.deltastreamer.current.source.name", topicName2);
+    properties.setProperty("hoodie.deltastreamer.current.source.checkpoint", topicName2 + ",0:0,1:0");
+    executionContexts.get(1).setProperties(properties);
+
+    TypedProperties properties1 = executionContexts.get(0).getProperties();
+    properties1.setProperty("hoodie.deltastreamer.schemaprovider.source.schema.file", dfsBasePath + "/source_short_trip_uber.avsc");
+    properties1.setProperty("hoodie.deltastreamer.schemaprovider.target.schema.file", dfsBasePath + "/target_short_trip_uber.avsc");
+    properties1.setProperty("hoodie.deltastreamer.source.kafka.topic", topicName1);
+    properties1.setProperty("hoodie.deltastreamer.current.source.name", topicName1);
+    properties1.setProperty("hoodie.deltastreamer.current.source.checkpoint", topicName1 + ",0:0,1:0");
+    executionContexts.get(0).setProperties(properties1);
+
+    String targetBasePath1 = executionContexts.get(0).getConfig().targetBasePath;
+    String targetBasePath2 = executionContexts.get(1).getConfig().targetBasePath;
+    streamer.sync();
+    // assert whether targetBasePath1 equals targetBasePath2 because that there is only one sink table.
+    assertEquals(targetBasePath1, targetBasePath2);
+    TestHoodieDeltaStreamer.TestHelpers.assertRecordCount(15, targetBasePath1 + "/*/*.parquet", sqlContext);
+
+    //insert updates for already existing records in kafka topics
+    testUtils.sendMessages(topicName1, Helpers.jsonifyRecords(dataGenerator.generateUpdatesAsPerSchema("001", 5, HoodieTestDataGenerator.TRIP_SCHEMA)));
+    testUtils.sendMessages(topicName2, Helpers.jsonifyRecords(dataGenerator.generateUpdatesAsPerSchema("001", 10, HoodieTestDataGenerator.SHORT_TRIP_SCHEMA)));
+
+    streamer = new HoodieMultiTableDeltaStreamer(cfg1, jsc);
+    streamer.getTableExecutionContexts().get(1).setProperties(properties);
+    streamer.getTableExecutionContexts().get(0).setProperties(properties1);
+    streamer.sync();
+
+    assertEquals(2, streamer.getSuccessTables().size());
+    assertTrue(streamer.getFailedTables().isEmpty());
+
+    //assert the record count matches now
+    TestHoodieDeltaStreamer.TestHelpers.assertRecordCount(15, targetBasePath1 + "/*/*.parquet", sqlContext);
+    testNum++;
+  }
+
+  @Test
+  public void testMultiTableExecutionWithKafkaSourceWhenMultiSinkTablesBoundMultiSources() throws IOException {
```

Comment:
In line with the difference of schema comment here - https://github.com/apache/hudi/pull/4925#issuecomment-1055454800, I feel it will be better to modify this test case to include the below scenario ->
1. sink_table1 bound to 2 topics
2. sink_table2 bound to a single topic
3. for sink_table1, let us have different schemas for the 2 source topics, I do not intend to have big differences in their schemas, absence of a single field in one of the schemas should suffice. Let us show the use case of matching the schemas to the target schema as discussed in the quoted comment. 

This should help users in understanding the kind of use cases covered in this PR. Later on we can either update the existing blog or write a new one to document this feature properly for everyone. Let me know in case of any doubts.

Reply:
Once the suggested changes are incorporated, we can actually remove `testMultiTableExecutionWithKafkaSourceWhenOneSinkTableBoundMultiSources` and have a single test case covering the entire scenario.